walnut/validate/validate.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package validate
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/walnut/schemata"
+
+	"github.com/xeipuuv/gojsonschema"
+	"gopkg.in/yaml.v2"
+)
+
+// CheckSchema accepts a filename and format then validate against the schema specified (either workflow or task)
+func CheckSchema(filename string, format string) {
+
+	rawYAML, err := ioutil.ReadFile(filename) // import YAML file
+	if err != nil {
+		fmt.Printf("ReadFile failed: %v", err)
+		os.Exit(1)
+	}
```

Comment:
I'd probably keep the file reading outside `CheckSchema` and pass the `rawYAML` as an argument instead of the filename. This way any errors coming from `CheckSchema` are related to the actual check and not to basic file opening.